docs/dapp/sapphire/quickstart.md

```diff
@@ -0,0 +1,395 @@
+decrypt the storage slot, so a default value is returned.
+
+Indeed, the gateway does not even have the key needed to decrypt the _key_ in
+the MKVS; it can tell _that_ a storage slot was written, but not which one
+(although it can make a very good guess by reading the contract code).
+So you can see that confidentiality is in effect, but it's not something
+end-users need to think too much about.
+
+[@oasisprotocol/sapphire-paratime]: https://www.npmjs.com/package/@oasisprotocol/sapphire-paratime
+
+## Create a Sapphire-Native dapp
+
+Porting an existing Eth app is cool, and can already provide benefits like
+protecting against MEV.
+However, starting from scratch with confidentiality in mind can unlock some
+really novel dapps and provide a [higher level of security].
+
+One simple-but-useful dapp that takes advantage of confidentiality is a
+[dead person's switch] that reveals a secret (let's say the encryption key to a
+data trove) if the operator fails to re-up before too long.
+Let's make it happen!
+
+[higher level of security]: guide.md#writing-secure-dapps
+[dead person's switch]: https://en.wikipedia.org/wiki/Dead_man's_switch
+
+### Init a new Hardhat project
+
+1. Make & enter a new directory
+2. `npx hardhat` then create a TypeScript project.
+3. Install `@nomicfoundation/hardhat-toolbox` and its peer dependencies.
+
+### Add the Sapphire Testnet to Hardhat
+
+Open up your `hardhat.config.ts` and drop in these lines.
+They should remind you a lot about what happened with Truffle.
+
+```diff
+diff --git a/hardhat.config.ts b/hardhat.config.ts
+index 414e974..49c95f9 100644
+--- a/hardhat.config.ts
++++ b/hardhat.config.ts
+@@ -3,6 +3,15 @@ import "@nomicfoundation/hardhat-toolbox";
+
+ const config: HardhatUserConfig = {
+   solidity: "0.8.9",
++  networks: {
++    sapphire: {
++      chainId: 0x5aff,
++      url: "https://testnet.sapphire.oasis.dev",
++      accounts: [
++        process.env.PRIVATE_KEY ?? Buffer.alloc(0, 32).toString("hex"),
++      ],
++    },
++  },
+ };
+
+ export default config;
+```
+
+### Get the Contract
+
+This is a Sapphire tutorial and you're already a Solidity expert, so let's not
+bore you with explaining the gritty details of the contract.
+Start by pasting [Vigil.sol] into `contracts/Vigil.sol`.
+
+While you're there, also place [run-vigil.ts] into `scripts/run-vigil.ts`.
+We'll need that later.
+
+[Vigil.sol]: https://github.com/oasisprotocol/sapphire-paratime/blob/main/examples/hardhat/contracts/Vigil.sol
+[run-vigil.ts]: https://github.com/oasisprotocol/sapphire-paratime/blob/main/examples/hardhat/scripts/run-vigil.ts
+
+#### Vigil.sol, the interesting parts
+
+The key state variables are.
+
+```solidity
+    SecretMetadata[] public _metas;
+    bytes[] private _secrets;
+```
+
+* `_metas` is marked with `public` visibility, so despite the state itself being
+  encrypted and not readable directly, Solidity will generate a getter that will
```

Comment:
```suggestion
  encrypted and not readable directly, **Solidity will generate a getter that will
```

Reply:
This isn't really that interesting for an experienced solidity dev. That's the whole point of `public` state vars, after all.